testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/StartTest.java

```diff
@@ -150,28 +151,50 @@ public class StartTest implements TestkitRequest {
 
     @Override
     public TestkitResponse process(TestkitState testkitState) {
-        return createResponse(COMMON_SKIP_PATTERN_TO_REASON);
+        TestkitResponse testkitResponse = createSkipResponse(COMMON_SKIP_PATTERN_TO_REASON);
+        if (testkitResponse != null) {
+            return testkitResponse;
+        }
+        if (StartSubTest.decidePerSubTest(data.getTestName())) {
+            return RunSubTests.builder().build();
+        }
+        return RunTest.builder().build();
     }
 
     @Override
     public CompletionStage<TestkitResponse> processAsync(TestkitState testkitState) {
-        TestkitResponse testkitResponse = createResponse(ASYNC_SKIP_PATTERN_TO_REASON);
+        TestkitResponse testkitResponse = createSkipResponse(ASYNC_SKIP_PATTERN_TO_REASON);
+        if (testkitResponse == null && StartSubTest.decidePerSubTestAsync(data.getTestName())) {
+            testkitResponse = RunSubTests.builder().build();
+        } else {
+            testkitResponse = RunTest.builder().build();
+        }
         return CompletableFuture.completedFuture(testkitResponse);
     }
 
     @Override
     public Mono<TestkitResponse> processRx(TestkitState testkitState) {
-        TestkitResponse testkitResponse = createResponse(REACTIVE_LEGACY_SKIP_PATTERN_TO_REASON);
+        TestkitResponse testkitResponse = createSkipResponse(REACTIVE_LEGACY_SKIP_PATTERN_TO_REASON);
+        if (testkitResponse == null && StartSubTest.decidePerSubTestReactiveLegacy(data.getTestName())) {
+            testkitResponse = RunSubTests.builder().build();
+        } else {
+            testkitResponse = RunTest.builder().build();
+        }
         return Mono.fromCompletionStage(CompletableFuture.completedFuture(testkitResponse));
     }
 
     @Override
     public Mono<TestkitResponse> processReactive(TestkitState testkitState) {
-        TestkitResponse testkitResponse = createResponse(REACTIVE_SKIP_PATTERN_TO_REASON);
+        TestkitResponse testkitResponse = createSkipResponse(REACTIVE_SKIP_PATTERN_TO_REASON);
+        if (testkitResponse == null && StartSubTest.decidePerSubTestReactive(data.getTestName())) {
+            testkitResponse = RunSubTests.builder().build();
+        } else {
+            testkitResponse = RunTest.builder().build();
+        }
         return Mono.fromCompletionStage(CompletableFuture.completedFuture(testkitResponse));
     }
 
-    private TestkitResponse createResponse(Map<String, String> skipPatternToReason) {
+    private TestkitResponse createSkipResponse(Map<String, String> skipPatternToReason) {
```

Comment:
I would consider returning `Optional<TestkitResponse>`. It should allow better logic composition in methods using it.